src/test/java/TestDataHandler.java

```diff
@@ -1,67 +1,78 @@
-import org.junit.Before;
+import org.junit.BeforeClass;
 import org.junit.Test;
 import org.mockito.Mockito;
 import training.supportbank.DataHandler;
+import training.supportbank.Transaction;
 
 import java.io.BufferedReader;
 import java.io.IOException;
 import java.math.BigDecimal;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
 import java.util.*;
 
 import static org.assertj.core.api.Assertions.*;
 
 public class TestDataHandler {
-    private static List<Map<String, String>> expectedArray = new ArrayList<>();
+    private static DataHandler dataHandler;
     private static Map<String, BigDecimal> expectedBalancesMap = new LinkedHashMap<>();
 
-    @Before
-    public void populateExpectedOutputs() {
-        expectedArray.clear();
-        expectedBalancesMap.clear();
+    private static Transaction testTransaction;
+    private static List<Transaction> testTransactionsList = new ArrayList<>();
 
-        Map<String, String> expectedMap = new LinkedHashMap<>();
-        expectedMap.put("Date", "01/01/14");
-        expectedMap.put("From", "John D");
-        expectedMap.put("To", "Jane D");
-        expectedMap.put("Narrative", "Doughnuts");
-        expectedMap.put("Amount", "5.00");
-        expectedArray.add(expectedMap);
+    @BeforeClass
```

Comment:
Moving this test setup into `@BeforeClass` still has the problem that this setup isn't relevant to all of the tests. Give me a shout and we'll have a chat about it. 🙂

Reply:
Tests moved following creation of new classes - refactored @BeforeClass methods too.